docs/sync-storage-handbook/src/data-model.md

```diff
@@ -0,0 +1,54 @@
+# Sync data model
+
+- Introduce high-level concepts here, and elaborate on the details elsewhere?
+
+## Data classes
+
+All synced data types fit into one of three broad categories.
+
+- Are there others we should think about?
+- Would a table comparing the different characteristics be useful?
+- These classifications aren't rigid; it'd be nice to find a way to describe that better.
+
+### Hierarchical data
+
+Bookmarks are an example of hierarchical data, or **trees**. Records in tree collections are interdependent: parents have pointers to their children, and children back to their parents.
+
+Hierarchical data present a problem for syncing, because records can be downloaded in any order. Additionally, some changes require multiple records to be uploaded in lockstep. Missing or incomplete records lead to problems like missing parents and children, and parent-child disagreements, which clients must resolve to ensure the server data remain consistent.
```

Comment:
nit: the word "data" [is complicated](https://www.quickanddirtytips.com/education/grammar/is-data-singular-or-plural) but I think you mean it as a singular mass noun here, so this should be "remains" rather than "remain".